jack.py

```diff
@@ -739,12 +739,21 @@ def set_process_callback(self, callback, userdata=None):
             is called.
 
         """
-        @self._callback("JackProcessCallback", error=STOP_CALLING)
-        def callback_wrapper(frames, _):
-            return callback(frames, userdata)
+        if isinstance(callback, _ffi.CData):
+            if _ffi.typeof(callback) != _ffi.typeof("JackProcessCallback"):
+                raise ValueError("Expected %s, got %s" % \
+                                 (_ffi.typeof("JackProcessCallback"),
+                                  _ffi.typeof(callback)))
```

Comment:
I was trying to avoid a CData that's not a JackProcessCallback from being considered as a python callable.